contrib/drivers/filesharedrivers/oceanstor/oceanclient.go

```diff
@@ -0,0 +1,503 @@
+	resp, err := c.request(url, "GET", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var fileSystem FileSystem
+	if err := handleReponse(resp, &fileSystem); err != nil {
+		return nil, err
+	}
+
+	return &fileSystem.Data, nil
+}
+
+func (c *Client) getFileSystemByName(name string) ([]FileSystemData, error) {
+	url := "/filesystem?filter=NAME::" + name
+	resp, err := c.request(url, "GET", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var fsList FileSystemList
+
+	if err := handleReponse(resp, &fsList); err != nil {
+		return nil, err
+	}
+
+	return fsList.Data, nil
+}
+
+func (c *Client) deleteFileSystem(name, poolID string) error {
+	data := map[string]interface{}{
+		"PARENTID":   poolID,
+		"NAME":       name,
+		"PARENTTYPE": 216,
+		"ALLOCTYPE":  1,
+	}
+
+	url := "/filesystem"
+	resp, err := c.request(url, "POST", data)
+	if err != nil {
+		return err
+	}
+
+	var fileSystem FileSystem
+
+	if err := handleReponse(resp, &fileSystem); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) ListStoragePools() ([]StoragePool, error) {
+	resp, err := c.request("/storagepool", "GET", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var pools StoragePoolList
+	if err := handleReponse(resp, &pools); err != nil {
+		return nil, err
+	}
+
+	return pools.Data, nil
+}
+
+func (c *Client) createSnapshot(fsID, snapName string) (*FSSnapshotData, error) {
+	data := map[string]string{
+		"PARENTTYPE":  "40",
+		"TYPE":        "48",
+		"PARENTID":    fsID,
+		"NAME":        strings.Replace(snapName, "-", "_", -1),
+		"DESCRIPTION": "",
+	}
+	url := "/FSSNAPSHOT"
+	resp, err := c.request(url, "POST", data)
+	if err != nil {
+		return nil, err
+	}
+
+	var fsSnapshot FSSnapshot
+	if err := handleReponse(resp, &fsSnapshot); err != nil {
+		return nil, err
+	}
+
+	return &fsSnapshot.Data, nil
+}
+
+func (c *Client) listSnapshots(fsID string) ([]FSSnapshotData, error) {
+	url := "/FSSNAPSHOT?sortby=TIMESTAMP,d&range=[0-100]&PARENTID=" + fsID
+	resp, err := c.request(url, "GET", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var fsSnapshotList FSSnapshotList
+	if err := handleReponse(resp, &fsSnapshotList); err != nil {
+		return nil, err
+	}
+
+	return fsSnapshotList.Data, nil
+}
+
+func (c *Client) deleteFSSnapshot(snapID string) error {
+	url := "/FSSNAPSHOT/" + snapID
+	resp, err := c.request(url, "DELETE", nil)
+	if err != nil {
+		return err
+	}
+
+	var errDelete DeleteError
+	if err := handleReponse(resp, &errDelete); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *Client) showFSSnapshot(snapID string) (*FSSnapshotData, error) {
+	url := "/FSSNAPSHOT/" + snapID
+	resp, err := c.request(url, "GET", nil)
+	if err != nil {
+		return nil, err
+	}
+
+	var fsSnapshot FSSnapshot
+	if err := json.Unmarshal(resp, &fsSnapshot); err != nil {
+		return nil, err
+	}
+
+	return &fsSnapshot.Data, nil
+}
+
+func (c *Client) getAllFilesystem() ([]FileSystemData, error) {
+	url := "/filesystem"
+	resp, err := c.request(url, "Get", nil)
```

Comment:
s/Get/GET

Reply:
Done. In the `c.request` method, the `GET`, `UPDATE` and so on are all changed to uppercase with strings.ToUpper(). The method name in this case has no effect.